actions/class.RestQtiTests.php

```diff
@@ -39,6 +41,29 @@ public function index()
         $this->returnFailure(new \common_exception_NotImplemented('This API does not support this call.'));
     }
 
+    /**
+     * Method will return qti package encoded in base64 for delivery
+     * @throws common_Exception
+     * @throws common_exception_Error
+     * @throws common_exception_NotImplemented
+     */
+    public function exportQtiPackage()
+    {
+        $params = $this->getPsrRequest()->getQueryParams();
+        if (!$deliveryId = $params[self::PARAM_TEST_URI]) {
+            $this->returnFailure(new common_exception_MissingParameter);
+        }
+        if (!$delivery = $this->getResource($deliveryId)) {
+            $this->returnFailure(new common_exception_ResourceNotFound);
+        }
+
+        $qtiPackage = $this->getQtiPackageExporter()->exportDeliveryQtiPackage($delivery);
```

Comment:
Doing it in the fly should causes http timeout, what do u think about using defferred task?